bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/tcc/ObservationEnvironment.java

```diff
@@ -126,7 +129,18 @@ public String getObservationTitle() {
     }
 
     public String getBasePositionName() {
-        return _targetEnv.getArbitraryTargetFromAsterism().getName();
+        final Asterism a = _targetEnv.getAsterism();
+
+        // A dual asterism always has a sky position, and an asterism with an
+        // overridden base has a sky position.
+        if (a instanceof GhostAsterism) {
```

Comment:
Do you need the external check? it seems only the internal check is needed? or maybe collapse it into a single condition

Reply:
It would have to be added to the internal check to ensure the conversion to `GhostAsterism` wouldn't fail; that's not problematic, but I thought it delineated the fact that the next `if` check was only applicable to `GhostAsterism`s better.